BEP174.md

```diff
@@ -32,50 +35,48 @@ After the implementation of this BEP:
-- Managers will have to register the 100BNB once their public key has been added to the set of Managers via governance
-- Once registered, Managers will be able to add an associated Relayer public key
+- Managers will NOT have to deposit any BNB (as opposed to 100BNB requirement which was before). 
+- Once selected by governance, Managers will be able to add an associated Relayer public key.
 - There will be a 1:1 relationship between Manager public key to active Relayer public key.
 - Only associated Relayers can send cross chain messages. 
-- Managers can modify public key of their Relayer at any time.  This will facilitate rotation of Relayer instances.
+- Managers can modify public key of their Relayer at any time. This will facilitate rotation of Relayer instances.
+- The old `unregister` function will still be active to allow the existing whitelisted relayers to safely exit.
 
 ### 5.3 `RelayerHub` Smart Contract Changes
 #### Relayer Manager functions
+
 ```
-function removeManagerByGov(address) external onlyGov
-```
-Removes Manager public key from current set of Managers. 100BNB will be returned to Manager address if the Manager has desposited.  Emits an event on success. This is only called via governance.
-```
-function removeManager() external onlyManager
-```
-Removes Manager public key from current set of Managers. 100BNB will be returned to Manager address if the Manager has desposited. Emits an event on success.
-```
-fuction addManagerByGov(address) external onlyGov
+function updateParam(string calldata key, bytes calldata value) external override onlyInit onlyGov
 ```
-Adds a Manager public key to the current set of Managers. This is only called via governance. Emits an event on success.
+The pre-existing function will be updated to support `addManager` and `removeManager` keys to facilitate adding and 
+removing of managers via governance.
 
 ```
-function registerManager() external payable onlyManager
+function removeManagerByHimself() external
 ```
-Registers an Manager and accepts 100BNB fee. Emits an event on success.
+Allows a manager to remove himself if required.
 
 ```
-function addRelayer(address) external onlyManager
+fuction addManagerByGov(address) external onlyGov
 ```
-Adds a Relayer for the calling Manager.  Overwrites Relayer if there is an existing Relayer already added. Emits an event on success.
+Adds a Manager public key to the current set of Managers. This is only called via governance. Emits an event on success.
 
 ```
-function registerManagerAddRelayer(address) external payable onlyManager
+function updateRelayer(address relayerToBeAdded) public onlyManager
 ```
-Calls both `registerManager` and `addRelayer`.
+This can be used by the manager to add or remove its relayer. Overwrites Relayer if there is an existing Relayer already
+added. Emits an event on success. The `relayerToBeAdded` parameter can be set to `0` address to enable removal of a 
+relayer (e.g. if the private key gets compromised).
```

Comment:
```suggestion
This can be used by the manager to add or remove its relayer. Overwrites Relayer if there is an existing Relayer already added. Emits an event on success. The `relayerToBeAdded` parameter can be set to `0` address to enable removal of a relayer (e.g. if the private key gets compromised).
```